workspaces/arborist/lib/shrinkwrap.js

```diff
@@ -979,8 +1003,10 @@ class Shrinkwrap {
     // if we don't have either, just an empty object so nothing matches below.
     // This will effectively just save the version and resolved, as if it's
     // a standard version/range dep, which is a reasonable default.
-    const spec = !edge ? rSpec
-      : npa.resolve(node.name, edge.spec, edge.from.realpath)
+    let spec = rSpec
+    if (edge) {
+      spec = npa.resolve(node.name, edge.spec, edge.from.realpath)
+    }
```

Comment:
```suggestion
    const spec = edge ? npa.resolve(node.name, edge.spec, edge.from.realpath) : rSpec
```

I 💯% appreciate the de-ternaryifying happenening in this PR. But that does make me want to call out that I think in this case the ternary is easier for me to read. I'm not asking for changes here, just showing my preference for the ternary operator in some cases.

Reply:
Would an if/else be better here? Is it the overloading of the `let` declaration?

Reply:
Rules are guidelines and should serve their purpose. If we find a case where ternaries are easier to read then let's do it.  Use them in *your* prs and let's find out.